fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/NonRdfSourcePersister.java

```diff
@@ -56,16 +65,98 @@ public NonRdfSourcePersister() {
     }
 
     @Override
-    public void persist(final OCFLObjectSession session, final ResourceOperation operation,
-                        final FedoraOCFLMapping mapping) throws PersistentStorageException {
+    public void persist(final PersistentStorageSession storageSession, final OCFLObjectSession session,
+                        final ResourceOperation operation, final FedoraOCFLMapping mapping)
+                                throws PersistentStorageException {
         log.debug("persisting ({}) to {}", operation.getResourceId(), mapping.getOcflObjectId());
         final String subpath = relativizeSubpath(mapping.getParentFedoraResourceId(), operation.getResourceId());
 
         // write user content
-        final NonRdfSourceOperation nonRdfSourceOperation = ((NonRdfSourceOperation) operation);
-        session.write(subpath, nonRdfSourceOperation.getContentStream());
+        final var nonRdfSourceOperation = (NonRdfSourceOperation) operation;
+        // TODO supply list of digests to calculate or wrap contentStream in DigestInputStream
+        final WriteOutcome outcome;
+        if (forExternalBinary(nonRdfSourceOperation)) {
+            outcome = null;
+        } else {
+            outcome = session.write(subpath, nonRdfSourceOperation.getContentStream());
+        }
+        // TODO verify digests in the outcome match supplied digests
```

Comment:
Is there a JIRA for this ?

Reply:
https://jira.lyrasis.org/browse/FCREPO-3148